### PR TITLE
scalability framework: upload regressions

### PR DESCRIPTION
**Pull request by nrainer-materialize**
Upload a `regressions.csv` file.

Tested successfully in https://buildkite.com/materialize/nightlies/builds/4784#018b4c91-c3a2-44a9-adf0-16e733bc9378.

### CODE REVIEW

**Comment by philip-stoev** [comment]
Can you add the name of the workload, and the containers/shas/mz_versiom()/git log outputs (whichever is readily available) as columns in the csv file? Right now if there are two regressed workloads they can not be differentiated in the .csv file.

**Comment by nrainer-materialize** [comment]
> Can you add the name of the workload, and the containers/shas/mz_versiom()/git log outputs (whichever is readily available) as columns in the csv file? Right now if there are two regressed workloads they can not be differentiated in the .csv file.

Sure!

File: test/scalability/mzcompose.py [hunk 1]
```diff
@@ -389,6 +394,18 @@ def create_result_analyzer(_args: argparse.Namespace) -> ResultAnalyzer:
...
 
+def upload_regressions_to_buildkite(outcome: RegressionOutcome) -> None:
+    if not outcome.has_regressions():
+        return
+
+    file_name = "regressions.csv"
+    file_path = RESULTS_DIR / file_name
+    outcome.raw_regression_data.to_csv(file_path)
+    spawn.runv(
+        ["buildkite-agent", "artifact", "upload", file_name], cwd=RESULTS_DIR
```
Comment (def-) [inline test/scalability/mzcompose.py:405]: You could also list the file in the pipeline file, then it will be picked up automatically.

**Review by def- — APPROVED** [review]

File: test/scalability/mzcompose.py (hunk 1 above)
Comment (nrainer-materialize) [reply test/scalability/mzcompose.py:405]: I will need to test if this works also when the file is located in some path.

**Comment by nrainer-materialize** [comment]
> Can you add the name of the workload, and the containers/shas/mz_versiom()/git log outputs (whichever is readily available) as columns in the csv file? Right now if there are two regressed workloads they can not be differentiated in the .csv file.

CSV content is now:
```
,count,concurrency,workload,baseline_tps,other_tps,tps_diff,tps_diff_perc,baseline_info,other_info
0,724,2,InsertWorkload,120.6757775933515,80.83941657371494,-39.836361019636556,-0.3301106635821777,v0.75.0-dev (50fb6882d),v0.66.0 (647282c5d)
```

**Pull request merged**